src/plugins/dashboard/public/application/legacy_app.ts

```diff
@@ -127,14 +140,67 @@ export function initDashboardApp(app, deps) {
           $scope.create = () => {
             history.push(DashboardConstants.CREATE_NEW_DASHBOARD_URL);
           };
+          $scope.dashboardItemCreators = () => {
+            return deps.dashboardItemCreators.map((creator) => ({
+              ...creator,
+              creatorFn: (event) => creator.creatorFn(event, history),
+            }));
+          };
+          $scope.dashboardItemCreatorClickHandler = (
+            creatorFn: DashboardCreatorFn
+          ): ((evt: any) => void) => {
+            return (evt: any) => {
+              creatorFn(evt, history);
+            };
+          };
+
+          $scope.editItemAvailable = (item) => !!item.editUrl;
```

Comment:
Was the typescript migration here necessary? While this file is now a typescript file, there are now many (36) type errors caused mostly due to the "no implicit any' rule. This makes the typescript migration not very useful. You aren't using any types here too.

Reply:
Ok.  Reverted.  

I was using "My" new types, for validation.  If it's not worth the lift, then ok.